src/bacnet/event.c

```diff
@@ -376,10 +450,190 @@ int event_notify_encode_service_request(
+                        apdu_len += len;
+
+                        len = encode_context_bitstring(&apdu[apdu_len], 1,
+                            &data->notificationParams.changeOfCharacterstring.statusFlags);
+                        apdu_len += len;
+
+                        len = encode_context_character_string(&apdu[apdu_len], 2,
+                            &data->notificationParams.changeOfCharacterstring.alarmValue);
+                        apdu_len += len;
+
+                        len = encode_closing_tag(&apdu[apdu_len], 17);
+                        apdu_len += len;
+                        break;
+
+                    case EVENT_CHANGE_OF_TIMER:
+                        len = encode_opening_tag(&apdu[apdu_len], 22);
+                        apdu_len += len;
+
+                        len = encode_context_enumerated(&apdu[apdu_len], 0,
+                            data->notificationParams.changeOfTimer.newState);
+                        apdu_len += len;
+
+                        len = encode_context_bitstring(&apdu[apdu_len], 1,
+                            &data->notificationParams.changeOfTimer.statusFlags);
+                        apdu_len += len;
+
+                        len = bacapp_encode_context_datetime(&apdu[apdu_len], 2,
+                            &data->notificationParams.changeOfTimer.updateTime);
+                        apdu_len += len;
+
+                        if (data->notificationParams.changeOfTimer.lastStateChange <= BACNET_TIMER_TRANSITION_EXPIRED_TO_RUNNING) {
+                            len = encode_context_enumerated(&apdu[apdu_len], 3,
+                                data->notificationParams.changeOfTimer.lastStateChange);
+                            apdu_len += len;
+                        }
+
+                        if (data->notificationParams.changeOfTimer.initialTimeoutSet) {
+                            len = encode_context_unsigned(&apdu[apdu_len], 4,
+                            data->notificationParams.changeOfTimer.initialTimeout);
+                            apdu_len += len;
+                        }
+
+                        if (data->notificationParams.changeOfTimer.expirationTimeSet) {
+                            len = bacapp_encode_context_datetime(&apdu[apdu_len], 5,
+                            &data->notificationParams.changeOfTimer.expirationTime);
+                            apdu_len += len;
+                        }
+
+                        len = encode_closing_tag(&apdu[apdu_len], 22);
+                        apdu_len += len;
+                        break;
                     case EVENT_EXTENDED:
-                    case EVENT_COMMAND_FAILURE:
+                    case EVENT_CHANGE_OF_STATUS_FLAGS:
+                    case EVENT_CHANGE_OF_RELIABILITY:
+                    case EVENT_CHANGE_OF_DISCRETE_VALUE:
                     default:
-                        assert(0);
+                        //assert(0);
```

Comment:
Is this meant to be commented out?

Reply:
yeah, dont want to kill the program if a unsupported event is attempted to be generated, will just take the line out